### ISSUE TITLE
[BUG] "Directory" field in "Custom document root" not load his state

### ISSUE THREAD

**Issue by fooger** [hestiacp/hestiacp#1844]
**Bug description:**
1) Open site editing
2) Set checkbox "Custom document root", select domain in "Point to" list and fill "Directory (Optional)" field.
3) Save. Custom document root is applied.
4) Open site editing again. The checkbox "Custom document root" is checked, but the "Directory (Optional)" field is empty. So, if I press "Save" again, Custom document root will changed to default (empty) state , and site stopped working.

Before saving:
![Before saving](https://i.imgur.com/vgLtUOF.png)

After saving (field "Directory" is empty, not loading his state):
![After saving](https://i.imgur.com/17P1qar.png)

After saving again (checkbox is unchecked, because in previously saving, I save the form with empty "Directory" field):
![After saving again](https://i.imgur.com/AnyvDHP.png)

**Hestia Versions:**
1.3.5 (fresh install, have this problem too), 1.4.1 (upgraded today)

**Operating system:**
Ubuntu 20.04

Fit ix please, thank you.

**Comment by jaapmarcus** [comment]
Does the custom_dir exists?

**Comment by fooger** [comment]
Yes, sure. After first saving (when I fill "Directory" field), site starting working from this dir. So, problem only in "loading saved directory to this field" when open the "Edit site" page.

**Comment by fooger** [comment]
```{
    "domain.com": {
        "IP": "xxx.x.xx.xxx",
        "IP6": "",
        "DOCUMENT_ROOT": "/home/domain/web/domain.com/public_html/public_html",
        "U_DISK": "1",
        "U_BANDWIDTH": "0",
        "TPL": "default",
        "ALIAS": "www.domain.com",
        "STATS": "",
        "STATS_USER": "",
        "SSL": "yes",
        "SSL_FORCE": "yes",
        "SSL_HSTS": "",
        "SSL_HOME": "same",
        "LETSENCRYPT": "yes",
        "FTP_USER": "",
        "FTP_PATH": "",
        "AUTH_USER": "",
        "BACKEND": "default",
        "PROXY": "default",
        "PROXY_EXT": "jpg,jpeg,gif,png,ico,svg,css,zip,tgz,gz,rar,bz2,exe,pdf,doc,xls,ppt,txt,odt,ods,odp,odf,tar,bmp,rtf,js,mp3,avi,mpeg,flv,html,htm",
        "FASTCGI_CACHE": "no",
        "FASTCGI_DURATION": "0s",
        "REDIRECT": "",
        "REDIRECT_CODE": "",
        "CUSTOM_DOCROOT": "/home/domain/web/domain.com/public_html/public_html",
        "SUSPENDED": "no",
        "TIME": "11:12:44",
        "DATE": "2021-04-15"
    }
}
```

public_html/public_html - it is not mistake. In domain dir, in public_html i have root files of framework, in subdirectory public_html (public_html/public_html) I have public files of site.

**Comment by jaapmarcus** [comment]
Oke thank you will look into what is going wrong it might be that we overwrite public_html via regex

**Comment by jaapmarcus** [comment]
Only issue is with public_html
array(3) { [0]=> string(49) "/home/xxxxx/web/xxxxx/public_html/build_html/" [1]=> string(7) "xxxx" [2]=> string(11) "build_html/" }

array(3) { [0]=> string(49) "/home/xxxxx/web/xxxxx/public_html/public_html/" [1]=> string(7) "xxxx/public_html"  }

**Comment by jaapmarcus** [comment]
https://github.com/hestiacp/hestiacp/pull/1854

Should fix the issue.  Have used: build_/html and other formats in the past was only able to replicate the issue with public_html for see PR for reasons

**Comment by fooger** [comment]
Nice, thank you very much!

**Comment by jaapmarcus** [comment]
Fixed for 1.4.2
